src/parser.rs

```diff
@@ -415,20 +415,19 @@ impl Library {
                                         "name" => field_name = attr.value.clone(),
                                         _ => {},
                                     }
+                                    break;
                                 }
 
                                 u = Union {
                                     name: format!(
-                                        "{}{}_{}{}",
-                                        parent_name_prefix.unwrap_or(""),
-                                        record_name,
+                                        "{}_{}{}",
+                                        parent_name_prefix.unwrap_or(record_name),
                                         field_name,
                                         union_count
                                     ),
                                     c_type: Some(format!(
-                                        "{}{}_{}{}",
-                                        parent_ctype_prefix.unwrap_or(""),
-                                        c_type,
+                                        "{}_{}{}",
+                                        parent_ctype_prefix.unwrap_or(c_type),
```

Comment:
That's true. You have to invent a name for it :) That's exactly why GTK is failing